src/passes/NoExitRuntime.cpp

```diff
@@ -46,7 +46,17 @@ struct NoExitRuntime : public WalkerPass<PostWalker<NoExitRuntime>> {
     }
     for (auto name : ATEXIT_NAMES) {
       if (name == import->base) {
-        replaceCurrent(Builder(*getModule()).replaceWithIdenticalType(curr));
+        // Remove the call, and drop the arguments (which may have side
+        // effects); let other passes clean that up more.
+        Builder builder(*getModule());
+        std::vector<Expression*> args;
+        for (auto* operand : curr->operands) {
```

Comment:
actually, do we have the EffectAnalyzer here? Would it make sense to actually throw away operands with no side effects? or will that happen in a later pass?

Reply:
That's an option, but yes, later passes do it anyhow, so it's not necessary. It would (very slightly, that's true) increase complexity here, so doesn't seem worth it I think.

Reply:
I think it's fine to rely on later passes for this.